atomicfu-gradle-plugin/src/main/kotlin/kotlinx/atomicfu/plugin/gradle/AtomicFUGradlePlugin.kt

```diff
@@ -42,15 +42,38 @@ private const val MIN_SUPPORTED_KGP_VERSION = "1.7.0"
 open class AtomicFUGradlePlugin : Plugin<Project> {
     override fun apply(project: Project) = project.run {
         checkCompatibility()
-        val pluginVersion = rootProject.buildscript.configurations.findByName("classpath")
-            ?.allDependencies?.find { it.name == "atomicfu-gradle-plugin" }?.version
+        // Get the version of atomicfu-gradle-plugin applied to the project:
+        // first try to find it in the root buildscript, then in the buildscript of the current project.
+        // The error is thrown in case atomicfu-gradle-plugin is not applied.
+        val pluginVersion = rootProject.getAGPVersion() ?: this.getAGPVersion()
```

Comment:
Does it work for projects having deeper subprojects layout?
```
<root project>
   <subproject A>
     <subproject A.1>
     <subproject A.2>
   <subproject B>
```

Reply:
Mm, probably, it will break if the current project is A.1 and `kotlinx-atomicfu` is applied in the buildscript of A.
Actually, we should walk through the whole project tree, I'm not sure, if there is an easier way to do that. 

I'll have a look 👀